presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1574,6 +1575,44 @@ public static ExpressionAnalysis analyzeExpression(
                 analyzer.getWindowFunctions());
     }
 
+    public static ExpressionAnalysis analyzeSqlFunctionExpression(
```

Comment:
It looks to me that this function should still belong to `SqlFunctionUtils.java`, despite some changes to this method.

Reply:
All function returning `ExpressionAnalysis` are static functions in `ExpressionAnalyzer`. I think `ExpressionAnalysis` is to wrap the data generated by `ExpressionAnalyzer` so the states are not exposed directly. If that's the case, this is the right place for the function, and other methods returning `ExpressionAnalyzer` should be changed to `private`.